app/webpacker/controllers/out_of_stock_modal_controller.js

```diff
@@ -0,0 +1,19 @@
+import { Controller } from "stimulus";
+
+// This is meant to be used with the "modal:closing" event, ie:
+//
+//  <div data-controller="out-of-stock-modal"
+//       data-action="moda:closing@out-of-stock-modal#redirect"
+//       data-out-of-stock-modal-redirect-value="true"
+//  >
+//  </div>
```

Comment:
As this is a specific-purpose controller already, can't we program it here to listen for the `closing` event on itself? Then we wouldn't need to configure it in the HTML.

Reply:
Or rather, why not override the `close` function? Something like:

``` 
close() {
   this.redirect();
   super();
}

Reply:
This controller doesn't actually inherit from the modal controller, but I think that something we could do. I'll give it a shot. 
The current solution is what I came up with before I decided to create a new component for the out of stock modal. 

> As this is a specific-purpose controller already, can't we program it here to listen for the closing event on itself? Then we wouldn't need to configure it in the HTML.

My understanding is this is the preferred way for cross controller communication : https://stimulus.hotwired.dev/reference/controllers#cross-controller-coordination-with-events
That means you don't have to worry about managing the liifecycle of event listener.

Reply:
I tried to inherit from the modal controller to do something like what David suggested, but it doesn't work because the `close()` function (and others) are added to the controller as mixin :
https://github.com/openfoodfoundation/openfoodnetwork/blob/96b8a4c0689c9c63030427ef82893a980aab3bd2/app/webpacker/controllers/modal_controller.js#L8-L13
The issue with this method of adding mixin is that it override the existing `close()` method, including the one created by the subclass. They are other method of adding mixin in js but I don't think it's worth the effort of changing here. See https://justinfagnani.com/2015/12/21/real-mixins-with-javascript-classes/ for more info.

Anyway I am going to leave the PR has it is.

Reply:
> This controller doesn't actually inherit from the modal controller

Sorry, I missed that detail, and assumed based on the name that it did. As it currently stands, it's not a "modal-controller", rather more like a "redirect-to-shop-controller". 

Thanks for giving it a shot.